### PR TITLE
Csharp models

### PR DESCRIPTION
**Pull request by biabeniamin**
I've completed the development of models generator in csharp.

### CODE REVIEW

File: DatabaseFunctionsGenerator/CSharpGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    public class CSharpGenerator : IGenerator
+    {
+        private Database _database;
+
+        private CSharpModelsGenerator _cSharpModelsGenerator;
+
+        public CSharpGenerator(Database database)
+        {
+            _database = database;
+
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/CSharpGenerator.cs]: Remove this spaces here. No need for them

File: DatabaseFunctionsGenerator/CSharpGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            _cSharpModelsGenerator = new CSharpModelsGenerator(_database);
+        }
+
+        public void Generate(string path)
+        {
+            string cSharpPath;
+
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/CSharpGenerator.cs]: Same as above.

File: DatabaseFunctionsGenerator/CSharpModelsGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            {
+                builder.AppendLine($"private {column.Type.GetCSharpType()} _{Helpers.GetLowerCaseString(column.Name)};");
+            }
+
+            foreach (Table parentTable in table.Parents)
+            {
+                builder.AppendLine($"private {parentTable.SingularName} _{parentTable.LowerCaseSingularName};");
+            }
+
+
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/CSharpModelsGenerator.cs]: same

File: DatabaseFunctionsGenerator/CSharpModelsGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+                {
+                    builder.AppendLine($"\t\t_{parentTable.LowerCaseSingularName} = value;");
+                }
+                builder.AppendLine("\t}");
+                builder.AppendLine("}");
+
+                builder.AppendLine();
+            }
+
+
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/CSharpModelsGenerator.cs]: same

File: DatabaseFunctionsGenerator/CSharpModelsGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+                //getter
+                builder.AppendLine($"public {parentTable.SingularName} {parentTable.SingularName}");
+                builder.AppendLine("{");
+                builder.AppendLine("\tget");
+                builder.AppendLine("\t{");
+                {
+                    builder.AppendLine($"\t\treturn _{parentTable.LowerCaseSingularName};");
+                }
+                builder.AppendLine("\t}");
+
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/CSharpModelsGenerator.cs]: same

File: DatabaseFunctionsGenerator/ColumnType.cs [hunk 6]
```diff
@@ -112,6 +112,23 @@ public string GetTypescriptType()
...
 
+        public string GetCSharpType()
+        {
+            switch (Type)
+            {
+                case Types.DateTime:
+                    return "DateTime";
+                case Types.Integer:
+                    return "int";
+                case Types.Text:
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/ColumnType.cs]: Does this work?

File: DatabaseFunctionsGenerator/Generator.cs [hunk 7]
```diff
@@ -95,19 +97,20 @@ public void Generate()
             AddMissingFields();
             SetParentChildsFields();
 
-            _sqlGenerator.Generate(path);
-            _phpGenerator.Generate(path);
-            _typescriptGenerator.Generate(path);
+            //_sqlGenerator.Generate(path);
+            //_phpGenerator.Generate(path);
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/Generator.cs]: Uncomment this part.

File: DatabaseFunctionsGenerator/Generator.cs [hunk 8]
```diff
@@ -95,19 +97,20 @@ public void Generate()
...
-            _sqlGenerator.Generate(path);
-            _phpGenerator.Generate(path);
-            _typescriptGenerator.Generate(path);
+            //_sqlGenerator.Generate(path);
+            //_phpGenerator.Generate(path);
+            //_typescriptGenerator.Generate(path);
+            _cSharpGenerator.Generate(path);
 
-            foreach (string file in Directory.EnumerateFiles($"{path}\\Php"))
+            /*foreach (string file in Directory.EnumerateFiles($"{path}\\Php"))
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/Generator.cs]: Why did you keep this commented part?

File: DatabaseFunctionsGenerator/MainWindow.xaml.cs [hunk 9]
```diff
@@ -71,9 +72,11 @@ public MainWindow()
 
             Database.Tables[4].Columns.Add(new Column("Pin", new ColumnType(Types.Varchar, 20)));
 
+
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/MainWindow.xaml.cs]: Useless spaces.

File: DatabaseFunctionsGenerator/CSharpModelsGenerator.cs [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+                //set
+                builder.AppendLine("\tset");
+                builder.AppendLine("\t{");
+                {
+                    builder.AppendLine($"\t\t_{column.LowerCaseName} = value;");
+                }
+                builder.AppendLine("\t}");
+                builder.AppendLine("}");
+
```
Comment (CosminNechifor) [inline DatabaseFunctionsGenerator/CSharpModelsGenerator.cs]: remove this line

File: DatabaseFunctionsGenerator/CSharpGenerator.cs (hunk 1 above)
Comment (biabeniamin) [reply DatabaseFunctionsGenerator/CSharpGenerator.cs]: I like to left some new lines between different type of code.

**Pull request merged**